torch/_refs/__init__.py

```diff
@@ -5744,21 +5744,52 @@ def log_normal(self, mean=1, std=2, generator=None):
 
 
 # TODO: add support for functionalization aten.normal_functional
+# NOTE: we cannot use the out wrapper because of normal.float_float_out
 @register_decomposition(aten.normal)
-@out_wrapper()
 @elementwise_type_promotion_wrapper(
     type_promoting_args=("self",),
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
-def normal(self, mean=0, std=1, generator=None):
+def normal(
+    mean=0,
+    std=1,
+    shape=None,
+    *,
+    generator=None,
+    out=None,
+    dtype=None,
+    layout=None,
+    device=None,
+    pin_memory=None,
+):
     assert generator is None
-    torch._check(std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}")
+    assert layout is None
+
+    if isinstance(std, Tensor):
+        shape = std.shape
+        dtype = std.dtype
+        device = std.device
+    else:
+        torch._check(
+            std >= 0, lambda: f"normal expects std >= 0.0, but found std {std}"
+        )
+
+    if isinstance(mean, Tensor):
+        shape = _broadcast_shapes(shape, mean.shape)
```

Comment:
What if `shape` is None?

Also, why do we just broadcast shapes in this path?

This logic is unnecessarily tricky I think.

Reply:
If `std` was a tensor the output shape is the results of broadcasting that shape with the mean. I think my code is still wrong in that case because I'm not computing the dtype correctly. `_broadcast_shapes` already filters out None types so adding a check whether shape is None seems redundent